Output logzio

idohalevi opened the pull request:
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

Commit:
update all.go with logz.io output

Commit:
add logzio output

Commit:
update main readme

Commit:
fix readme

ofir76 commented on plugins/outputs/logzio/logzio.go:
```diff
@@ -0,0 +1,165 @@
+}
+
+func (l *Logzio) newLogzioSender() error {
+	if l.Token == "" || l.Token == "your logz.io token" {
```
the default can't be nil?

ofir76 commented on plugins/outputs/logzio/logzio.go:
```diff
@@ -0,0 +1,165 @@
+	)
+
+	if err != nil {
+		return fmt.Errorf("[logzio] failed to create new logzio sender: %s", err)
```
i thing you need to change %s to %v for printing all the err

ofir76 commented on plugins/outputs/logzio/logzio.go:
```diff
@@ -0,0 +1,165 @@
+		return fmt.Errorf("[logzio] failed to create new logzio sender: %s", err)
+	}
+
+	log.Printf("I! [logzio] Successfuly created Logz.io sender: %s %s %s %d\n", l.URL, l.QueueDir,
```
%s with int? you need to change %d

ofir76 commented on plugins/outputs/logzio/logzio.go:
```diff
@@ -0,0 +1,165 @@
+	sender *lg.LogzioSender
+}
+
+func (l *Logzio) newLogzioSender() error {
```
maybe initializeSender its better?

ofir76 commented on plugins/outputs/logzio/logzio.go:
```diff
@@ -0,0 +1,165 @@
+		}
+		err = l.sender.Send(serialized)
+		if err != nil {
+			return fmt.Errorf("E! [logzio] Failed to send metric: %s\n", err)
```
%s and error need to be %v and error

ofir76 commented on plugins/outputs/logzio/logzio.go:
```diff
@@ -0,0 +1,165 @@
+	return nil
+}
+
+func NewLogzioOutput() *Logzio {
```
CreateDefultLogizoOutput

Commit:
renaming methods

Commit:
fixing logzio file path

Commit:
fix readme

Commit:
adding deleted rows in the readme file

Commit:
remove logzio-go use

Commit:
adding tests

Commit:
fix URL to include token in the request

Commit:
fmt

Commit:
fix tests

Commit:
fix readme

Commit:
changes based on CR

Commit:
update readme and sampleConfig with tls options

ssoroka committed:
execd processor

ssoroka committed:
Shim refactor to support processors and output

danielnelson committed:
Update changelog

danielnelson committed:
Update to github.com/shirou/gopsutil v2.20.5 (#7641)

danielnelson committed:
Update changelog

danielnelson committed:
Update changelog

sboeckelmann committed:
Change rpm dist packaging type for arm64 to aarch64 (#7645)

haidlir committed:
Add timezone configuration to csv data format (#7619)

danielnelson committed:
Update changelog

essobedo committed:
Add ability to collect response body as field with http_response (#7596)

danielnelson committed:
Update changelog

danielnelson committed:
Update changelog

danielnelson committed:
Set 1.14.4 release date

ssoroka committed:
link to glob pattern docs (#7657)

ssoroka committed:
Flaky shim test (#7656)

danielnelson committed:
Fix race issue in tick_test.go (#7663)

Trovalo committed:
Improve sqlserver input compatibility with older server versions (#7495)

danielnelson committed:
Update changelog

danielnelson committed:
Remove trailing backslash from tag keys/values (#7652)

danielnelson committed:
Update changelog

sjwang90 committed:
Update gNMI plugin readme (#7685)

essobedo committed:
Fix processor initialization (#7693)

danielnelson committed:
Mark unused agent options as deprecated

ssoroka committed:
procstat performance enhancement (#7686)

up to 40/120x better performance on FullPattern/Pattern functions

ssoroka committed:
update CHANGELOG.md

rawkode committed:
Add state and readiness to kube_inventory pod metrics (#7691)

danielnelson committed:
Fail check-deps when differences are found (#7694)

scmi1 committed:
Add 'batch' to mqtt output optional parameters (#7690)

danielnelson committed:
Link to GJSON playground in json parser documentation (#7698)

Trovalo committed:
Remove trailing backslash management in sqlserver input (#7700)

Backslash removal on tags is performed by the influx serializer now.

danielnelson committed:
Document that string fields do not produce prometheus metrics (#7644)

jimmyseto committed:
Add ability to add selectors as tags in kube_inventory (#7267)

danielnelson committed:
Update changelog

sarvanikonda committed:
Add redfish input plugin (#7082)

danielnelson committed:
Update changelog and redfish docs

nferch committed:
Fix ping exit code handling on non-Linux (#7658)

danielnelson committed:
Update changelog

benzvan committed:
Remove master/slave terminology from tests (#7719)

ribalda committed:
Skip overs errors in the output of the sensors command (#7718)

danielnelson committed:
Update changelog

denzilribeiro committed:
Add counter type to perfmon collector (#7712)

danielnelson committed:
Update changelog

piter75 committed:
Add missing nvme attributes to smart plugin (#7575)

danielnelson committed:
Update changelog

danielnelson committed:
Add starlark processor (#7660)

danielnelson committed:
Update readme/changelog

danielnelson committed:
Fix license check

ssoroka committed:
remove processors/execd/examples/count.go

ssoroka committed:
fix after rebase

ssoroka committed:
fix race

denzilribeiro committed:
Fix incorrect Azure SQL DB server properties (#7715)

danielnelson committed:
Update changelog

reimda committed:
Fix SNMP trap test race (#7731)

danielnelson committed:
Fix data race in kafka_consumer_test.go (#7737)

programmer04 committed:
Fix data race in plugins/inputs/suricata/suricata_test.go (#7745)

programmer04 committed:
Fix data race in plugins/inputs/stackdriver/stackdriver_test.go (#7744)

danielnelson committed:
Add laundry to mem input plugin on FreeBSD (#7736)

Additionally only report fields that may be set on the platform.

danielnelson committed:
Update changelog

danielnelson committed:
Export internal/tls package for use in execd plugins (#7697)

danielnelson committed:
Update tls import path

danielnelson committed:
Return on toml parse errors instead of logging (#7751)

danielnelson committed:
Update changelog

ssoroka committed:
address feedback

ssoroka committed:
clean up tests

ssoroka committed:
remove streaming processors docs

ssoroka committed:
Execd processor (#7640)

ssoroka committed:
reverse dns lookup processor (#7639)

danielnelson committed:
Update github.com/tidwall/gjson (#7756)

danielnelson committed:
Update changelog

danielnelson committed:
Build packages in makefile (#7759)

danielnelson committed:
Allow any key usage type on x509 certificate (#7760)

danielnelson committed:
Update release notes

danielnelson committed:
Update changelog

danielnelson committed:
Only set version ldflags on tags

ad-m committed:
Fix typo in elasticsearch input docs (#7764)

danielnelson committed:
Allow histograms with no buckets and summary without quantiles (#7740)

danielnelson committed:
Update changelog

danielnelson committed:
Set 1.14.5 release date

danielnelson committed:
Fix data race in phpfpm initializing http client (#7738)

danielnelson committed:
Allow overriding the collection_jitter and precision per input (#7762)

danielnelson committed:
Update changelog

ssoroka committed:
Rename cisco_telemetry_gnmi input to gnmi (#7695)

danielnelson committed:
Update readme and changelog

Commit:
Add nginx_sts input plugin (#7205)

danielnelson committed:
Update changelog

danielnelson committed:
Update nginx_sts plugin readme

danielnelson committed:
Update common/tls import path

krzysdabro committed:
Accept decimal point when parsing kibana uptime (#7768)

danielnelson committed:
Update changelog

danielnelson committed:
Set user agent when requesting http config (#7752)

ssoroka committed:
execd output (#7761)

ssoroka committed:
Update CHANGELOG.md

programmer04 committed:
Fix data race in tail input tests (#7780)

programmer04 committed:
Fix data race in plugin output pubsub tests (#7782)

ssoroka committed:
Fix flakey processors.execd test

programmer04 committed:
Run all Go tests with flag -race

programmer04 committed:
Do not enable -race for GOARCH=386

ssoroka committed:
switch mac tests to Go 1.14 (#7784)

ssoroka committed:
streaming processors docs update (#7786)

programmer04 committed:
Fix data race in input plugin ping_windows

ssoroka committed:
Fix inputs.execd readme links (#7791)

sergeykhegay committed:
Add v3 metadata support to ecs input (#7154)

danielnelson committed:
Update changelog

ssoroka committed:
Run all Go tests with flag -race (#7783)

danielnelson committed:
Support utf-16 in file and tail inputs (#7792)

danielnelson committed:
Update changelog

danielnelson committed:
Fix test race in kafka_consumer (#7797)

danielnelson committed:
Traverse redfish api using resource links (#7722)

reimda committed:
Add ifname processor plugin (#7763)

reimda committed:
Update changelog

danielnelson committed:
Update readme and changelog

danielnelson committed:
Update sample configuration

danielnelson committed:
Update next version

danielnelson committed:
Fix tag package version

sjwang90 committed:
Fix typo in 1.15 release notes (#7804)

ykzts committed:
Fix markdown syntax (#7806)

danielnelson committed:
Add minimum version for new plugins (#7810)

eraac committed:
fix typo in cloud_pubsub docs (#7815)

eraac committed:
documentation: typo outputs cloud_pubsub (#7817)

danielnelson committed:
Close file to ensure it has been flushed (#7819)

danielnelson committed:
Fix darwin package build flags (#7818)

ssoroka committed:
fix bug with loading plugins in shim with no config (#7816)

danielnelson committed:
Set log output before starting plugin (#7820)

ykzts committed:
Add github.com/inabagumi/youtube-telegraf-plugin to list of external plugins (#7825)

PierreF committed:
Send metrics in FIFO order (#7814)

Metrics are send from older to newer metrics, even when outputs is
failing.  In case of buffer full, we still drop the oldest metrics, but
non-dropped metrics are send in the order they are received.

pierwill committed:
Edit Starlark README (#7832)

ssoroka committed:
update go versions: 1.14.5, 1.13.13 (#7837)

reimda committed:
ifname processor: expire old cached entries (#7838)

danielnelson committed:
Update telegraf.conf

sjwang90 committed:
Update redfish docs with link (#7846)

danielnelson committed:
Initialize aggregation processors (#7853)

danielnelson committed:
Log after interval has elapsed; skip short intervals (#7854)

reimda committed:
ifname: avoid unpredictable conditions in getMap test (#7848)

kiokoman committed:
Fix suricata input docs (#7856)

trashhalo committed:
prefer testify error funcs over nil checks (#7857)

danielnelson committed:
Print configuration for downloads page (#7862)

ssoroka committed:
clarify docs and add warning if execd is misconfigured (#7866)

sjwang90 committed:
Add logic starlark example (#7864)

ssoroka committed:
fix issue with execd restart_delay being ignored (#7867)

danielnelson committed:
Update docker FAQ (#7868)

danielnelson committed:
Recv next message after send returns EOF (#7872)

sjwang90 committed:
Fix defaults processor readme typos (#7873)

ssoroka committed:
shim logger improvements (#7865)

reimda committed:
Set 1.15.0 release date

danielnelson committed:
Fix arch name in deb/rpm builds (#7877)

danielnelson committed:
Update changelog

DEvil0000 committed:
fixed cgroups docu (#7889)

ssoroka committed:
jolokia: add some error message context (#7906)

sensor-freak committed:
Fix/extend support of fixed point values on input (modbus plugin) (#7869)

* Add input float types FIXED/UFIXED (#7317)

vikkyomkar committed:
MongoDB: Added new metric "pages written from cache" (#7870)

ssoroka committed:
fixes issue with rpm /var/log/telegraf permissions (#7909)

matse committed:
fix broken link to proc.c (#7918)

reimda committed:
Fix tail following on EOF (#7927)

reimda committed:
Update changelog

kidk committed:
Fixed title on New Relic README.md (#7948)

kelseiv committed:
Add descriptions for measurement & fields (#7902)

Aladex committed:
JSON does not support values NaN and Inf (#7908)

alsargent committed:
Add details to connect to InfluxDB OSS 2 and Cloud 2 (#7953)

ssoroka committed:
cleanup cloudwatch plugin (#7928)

vipinvkmenon committed:
Support for AWS Cloudwatch Alarms #7931 (#7932)

This documentation update points to a simple execd shim that can
poll AWS Alarms at given time intervals.

ssoroka committed:
fix cloudwatch tests

Trovalo committed:
added new counter - Lock Timeouts (timeout > 0)/sec (#7808)

pabigot committed:
rabbitmq: remove deprecated healthcheck (#7823) (#7920)

russorat committed:
fix(readmes): adding code block annotations (#7963)

vikkyomkar committed:
Updated http_response input plugin README.md (#7962)

russorat committed:
fix(readmes): standarize first line of readmes (#7973)

jsteenb2 committed:
chore: extend gitignore to ignore .DS_Store (#7974)

russorat committed:
fix(ipmi): update link in readme (#7975)

russorat committed:
fix(puppet): update broken link (#7977)

samhld committed:
add pivot example for starlark processor (#7976)

Slach committed:
add additional metrics to clickhouse input plugin (#7904)

add metrics and test for following ClickHouse tables:
- system.zookeeper
- system.detached_parts
- system.dictionaries
- system.replication_queue
- system.mutations
- system.disks
- system.processes
- system.text_log
- system.part_log

thschue committed:
Dynatrace output plugin (#7881)

Commit:
Proxmox plugin (#7922)

ssoroka committed:
Update Bug_report.md

ssoroka committed:
add label to feature request.

ssoroka committed:
Create Support.md

ssoroka committed:
Update Support.md

russorat committed:
fix(readmes): updates to internal and proxmox readmes (#7982)

ssoroka committed:
http_listener_v2: make http header tags case insensitive (#7986)

ssoroka committed:
Update CHANGELOG.md

thschue committed:
Dynatrace Plugin - Updated Documentation (#7994)

ssoroka committed:
fix flakey ticker test (#7997)

ssoroka committed:
fix minor agent race condition around error messages (#7999)

ssoroka committed:
fix issue with shim use of config.Duration (#7996)

Diabl0Man committed:
[outputs.application_insights] Added the ability to set the endpoint url (#7134)

Hipska committed:
[agent] Cleanup duplicates in loaded plugins list (#7852)

sjwang90 committed:
Fix configuration.md formatting (#7965)

* fix configuration.md formatting

* include cloud url comments

sjwang90 committed:
add example input/outputs to starlark examples (#7980)

* add sample input/outputs to starlark examples

* update ratio.star with example input/outputs

* Update rename.star

* update scale.star with input/output examples

* fix spacing

* fix spacing

DasSkelett committed:
Fix Ping Input plugin for FreeBSD's ping6 (#7861)

thschue committed:
Fix InsecureSkipVerify in Dynatrace Plugin (#8015)

ssoroka committed:
improve the quality of starlark docs by executing them as tests (#8020)

pmalek-sumo committed:
Fix string to int conversion in eventhub consumer (#8006)

prydin committed:
vSphere Fixed missing clustername issue 7878 (#8026)

Hipska committed:
Add addTag debugging in ifname plugin (#8018)

sjwang90 committed:
Create external plugins doc file and update contributing guide (#8001)

ssoroka committed:
docs fix

grossjo committed:
Add set, and histogram reporting to aerospike telegraf plugin (#8025)

Co-authored-by: Joshua Gross <joshua.gross@indexexchange.com>

ssoroka committed:
fix docker build. update dockerfiles to Go 1.14 (#8051)

ssoroka committed:
fix bug in shim logger affecting AddError (#8052)

srebhan committed:
Fix CI AppVeyor 'make check' errors for go 1.15 on windows (#8061)

Avoid writing to nul on windows.  On Appveyor, writing to nul creates a file instead of discarding output.  The file's existence creates errors later in the build.

srebhan committed:
Update gopsutils to v.2.20.7 (#8073)

Also update to the specific version of golang.org/x/sys that gopsutils requires.

chrishayles committed:
Add OPC UA input plugin (#8009)

pmalek-sumo committed:
Fix docker-image make target (#8005)

sjwang90 committed:
add guidelines run to external plugins with execd (#8054)

* docs - add guidelines run to external plugins with execd

sjwang90 committed:
add tagpass/tagdrop note in docs (#8053)

sjwang90 committed:
update opcua readme (#8075)

reimda committed:
Opcua cleanup (#8076)

pmalek-sumo committed:
Fix external plugin docs links (#8081)

gearnode committed:
add nsd input plugin (#7822)

jdef committed:
[fix] add missing error check for HTTP req failure (#8071)

Commit:
added force_document_id option to ES output enable resend data and avoiding duplicated ES documents, fix #7891 (#8019)

denzilribeiro committed:
Database_type config to Split up sql queries by engine type (#7934)

sjwang90 committed:
update influxdb v2 port

ssoroka committed:
Revert "update influxdb v2 port"

This reverts commit 8c3ba65e77ea0d1af9875b2035e7d2f5051c6292.

pmalek-sumo committed:
Sumo Logic output plugin (#8023)

pmalek-sumo committed:
Carbon2 configuration option - include field in metric name (#8094)

hackery committed:
SNMP input plugin: document partial table collection (#8104)

Co-authored-by: James Beckett <hackery@users.noreply.github.com>

sjwang90 committed:
update 1.16 changelog with new plugins (#8107)

sjwang90 committed:
Fix bug fix typo in changelog (#8117)

magichair committed:
Influxdb v2 listener (#7828)

sspaink committed:
Doc update: Fix main.go shim example link (#8123)

Commit:
Sort plugin name lists for output (#8127)

sjwang90 committed:
add influxdata slack icon to telegraf readme (#8126)

sspaink committed:
Add octoprint to list (#8131)

eraac committed:
influxdb_v2_listener - add /ready route (#8133)

samm-git committed:
Remove Event field as it is causing issues with pre-trained source types (#8039)

Co-authored-by: Oleksii Samorukov <oleksii_samorukov@mckinsey.com>

M0rdecay committed:
Increasing the metric buffer (#8145)

ssoroka committed:
update shim docs to point users in the right direction

ssoroka committed:
minor docs cleanup

ssoroka committed:
Starlark readme url fix

Commit:
Changing duration condition and rebasing code

Commit:
git pull

Commit:
Fixing rebase

Commit:
Change imports to fix test
